### PR TITLE
fix: show speaker in menu when speker is available

### LINKED ISSUE #4999: Public Event Pages: Speakers Page Link even though No Speakers are Visible on Page
There is a speakers page on some events even if there is no speaker defined, e.g. compare https://eventyay.com/e/5ac8e4d9/speakers and https://eventyay.com/e/c310c6ba.

Expected:
* Only show "Speaker Link" if speakers are defined
* Only show speaker page once accepted or confirmed sessions exist.

Screenshot from 2020-09-07 15-45-02

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #4999

scrnli_9_22_2020_7-11-13 PM

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by mariobehling — CHANGES_REQUESTED** [review]
Yes, good. I see the speakers link is no longer in the sidebar. 

The speakers page should also not exist or not be accessible if there is no speaker, e.g. https://open-event-frontend-git-fork-maze-runnar-showspeaker.eventyay.vercel.app/e/5ac8e4d9/speakers

**Comment by maze-runnar** [comment]
> The speakers page should also not exist or not be accessible

@iamareebjamal  how to do it?  No such feature is avail for /schedule or /calendar is available on event page.

File: app/templates/components/public/side-menu.hbs [hunk 1]
```diff
@@ -9,7 +9,7 @@
           {{t 'Tickets'}}
         </a>
       {{/if}}
-      {{#if this.showSpeakers}}
+      {{#if (and this.showSpeakers this.event.speakers.length)}}
```
Comment (iamareebjamal) [inline app/templates/components/public/side-menu.hbs:12]: This is not good. Now it'll load speakers on the index page itself. Check the problem in showSpeakers and fix that instead

File: app/templates/components/public/side-menu.hbs [hunk 2]
```diff
@@ -23,7 +23,7 @@
...
```
Comment (iamareebjamal) [inline app/templates/components/public/side-menu.hbs:26]: Same here

**Comment by maze-runnar** [comment]
`await this.loader.load(`/events/${this.event.id}/speakers?fields[speaker]=id&page[size]=1&filters=${JSON.stringify(SPEAKERS_FILTER)}`)).data.length`

in this condition the above code is not working. even Schedule is not showing in side tab due to ` isSchedulePublished` condition, showSession is not working AS expected there too.

**Comment by maze-runnar** [comment]
can I use isSessionScheduled in Speaker Section too?

**Comment by maze-runnar** [comment]
`await this.loader.load(`/events/${this.event.id}/speakers?fields[speaker]=id&page[size]=1&filters=${JSON.stringify(SPEAKERS_FILTER)}`)).data.length > 1`

File: app/components/public/side-menu.js [hunk 3]
```diff
@@ -24,7 +23,7 @@ export default class SideMenu extends Component {
   }
 
   async checkSpeakers() {
-    this.showSpeakers = this.showSpeakers || (await this.loader.load(`/events/${this.event.id}/speakers?fields[speaker]=id&page[size]=1&filters=${JSON.stringify(SPEAKERS_FILTER)}`)).data.length;
+    this.showSpeakers = this.showSpeakers || this.event.speakers.length;
```
Comment (iamareebjamal) [inline app/components/public/side-menu.js:26]: As I said, this is not a good idea as it loads all speakers in one go. Secndly, it does not filter the speakers who have no accepted or confirmed sessions. It was used for a reason

Comment (maze-runnar) [reply app/components/public/side-menu.js:26]: trying to figure out why it's happening there.

Comment (maze-runnar) [reply app/components/public/side-menu.js:26]: https://api.eventyay.com//v1/events/1211/speakers and https://api.eventyay.com//v1/events/5ac8e4d9/speakers they both are on same page.
first one is returning empty response and second is giving response

Comment (iamareebjamal) [reply app/components/public/side-menu.js:26]: There must be some discrepancies in filters used

**Comment by iamareebjamal** [comment]
Superseded by #5171

**Pull request closed without merging**